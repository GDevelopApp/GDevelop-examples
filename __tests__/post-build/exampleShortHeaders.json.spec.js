// @ts-check
const fs = require('fs').promises;
const path = require('path');

/** @typedef {import('../../scripts/types').ExampleShortHeader} ExampleShortHeader */

/** @return {Promise<ExampleShortHeader[]>} */
const getExampleShortHeaders = async () => {
  const exampleShortHeadersJson = await fs.readFile(
    path.join(__dirname, '../../dist/database/exampleShortHeaders.json')
  );
  const exampleShortHeaders = JSON.parse(exampleShortHeadersJson.toString());
  return exampleShortHeaders;
};

describe('exampleShortHeaders.json post build checks', () => {
  test('defaultTags', async () => {
    const exampleShortHeaders = await getExampleShortHeaders();

    // Check that the headers seem correct
    expect(exampleShortHeaders.length).toBeGreaterThan(120);

    const platformer = exampleShortHeaders.find(
      ({ name }) => name === 'Platformer'
    );
    if (!platformer)
      throw new Error('Platformer not found in the example short headers');

    expect(platformer).toEqual(
      expect.objectContaining({
        id: '244c50f590aaef77fda8c883e95e64c728fc75be0825e42093b275b56320667f',
        slug: 'platformer',
        name: 'Platformer',
        authorIds: ['R0F5QGNCzgOY5w2cxGeKJOq2UaD2'],
        shortDescription:
          'An example of a basic platformer (Mario-like) game. Jump around and collect as many coins as you can!',
        license: 'MIT',
        previewImageUrls: [
          'https://resources.gdevelop-app.com/examples/platformer/preview.png',
          'https://resources.gdevelop-app.com/examples/platformer/square-icon.png',
        ],
        tags: expect.arrayContaining([
          'platformer',
          'Starter',
          'Mathematical tools',
          'Platform behavior',
          'Gamepads (controllers)',
        ]),
        gdevelopVersion: '',
      })
    );
  });
});
