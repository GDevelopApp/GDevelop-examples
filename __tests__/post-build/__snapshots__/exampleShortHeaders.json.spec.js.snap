// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exampleShortHeaders.json post build checks defaultTags 1`] = `
Object {
  "authorIds": Array [
    "R0F5QGNCzgOY5w2cxGeKJOq2UaD2",
  ],
  "codeSizeLevel": "small",
  "description": "Check out [the YouTube video](https://www.youtube.com/watch?v=eU0kkLSdw0Y&list=PL3YlZTdKiS898Wio0tvKjQM0x3zo4V0Mb) explaining how this example was made.",
  "difficultyLevel": "simple",
  "gdevelopVersion": "",
  "id": "d260466ba96262fd00930eb950ae486209a828f851add1c1c5bbc2a2d020f868",
  "license": "MIT",
  "name": "Platformer",
  "previewImageUrls": Array [
    "https://resources.gdevelop-app.com/examples/platformer/preview.png",
    "https://resources.gdevelop-app.com/examples/platformer/square-icon.png",
  ],
  "shortDescription": "An example of a basic platformer (Mario-like) game. Jump around and collect as many coins as you can!",
  "slug": "platformer",
  "tags": Array [
    "platformer",
    "game",
    "simple",
    "Starter",
    "Anchor",
    "Objects with animations",
    "Event functions",
    "Sounds and music",
    "Layers and cameras",
    "Events and control flow",
    "External layouts",
    "Mathematical tools",
    "Objects",
    "Scene",
    "Timers and time",
    "Variables",
    "Debugger Tools",
    "Objects with effects",
    "Flippable objects",
    "Linked objects",
    "Objects with opacity",
    "Panel Sprite (9-patch) Object",
    "Particle system",
    "Platform behavior",
    "Shape painter",
    "Resizable objects",
    "Scalable objects",
    "Sprite",
    "System information",
    "Objects containing a text",
    "Text object",
    "Tiled Sprite Object",
    "Tweening",
    "Parallax for Tiled Sprite",
    "Panel sprite button",
    "Platformer character animator",
    "Rectangular movement",
    "Smooth Camera",
    "Checkpoints",
    "Gamepads (controllers)",
    "Multitouch joystick and buttons (sprite)",
    "Shake object",
    "Volume Falloff",
    "Screen Orientation Checker",
  ],
}
`;

exports[`exampleShortHeaders.json post build checks defaultTags 2`] = `
Object {
  "authorIds": Array [
    "jy7FXnGX0ZZcWfrAI9YuQaeIphi1",
    "9MGDlUQAh8QUilno4JPycekjRCJ3",
    "R0F5QGNCzgOY5w2cxGeKJOq2UaD2",
  ],
  "codeSizeLevel": "small",
  "description": "",
  "difficultyLevel": "simple",
  "gdevelopVersion": "",
  "id": "3cfc75d79ece13652857e8f9c20931f275916ca88f89940ca81b91734da4ac5d",
  "license": "MIT",
  "name": "Tappy plane",
  "previewImageUrls": Array [
    "https://resources.gdevelop-app.com/examples/tappy-plane/thumbnail.png",
  ],
  "quickCustomizationImageUrl": "https://resources.gdevelop-app.com/examples/tappy-plane/thumbnail-quick-customization.png",
  "shortDescription": "Tappy Plane is a Flappy Bird-like game. Tap to keep your plane in the air and make your way through the cave system.
Get as far as you can before you crash in to a wall, submit your score, and try again.",
  "slug": "tappy-plane",
  "tags": Array [
    "tappy-plane",
    "game",
    "simple",
    "Starter",
    "Anchor",
    "Objects with animations",
    "Event functions",
    "Sounds and music",
    "Layers and cameras",
    "Events and control flow",
    "External layouts",
    "Storage",
    "Keyboard",
    "Mathematical tools",
    "Mouse and touch",
    "Network",
    "Objects",
    "Scene",
    "Timers and time",
    "Variables",
    "Destroy Outside Screen Behavior",
    "Objects with effects",
    "Flippable objects",
    "Leaderboards",
    "Objects with opacity",
    "Panel Sprite (9-patch) Object",
    "Platform behavior",
    "Player Authentication",
    "Resizable objects",
    "Scalable objects",
    "Sprite",
    "Objects containing a text",
    "Text Input",
    "Text object",
    "Tiled Sprite Object",
    "Tweening",
    "Ellipse movement",
    "Panel sprite button",
    "Shake object",
    "Flash object",
  ],
}
`;

exports[`exampleShortHeaders.json post build checks defaultTags 3`] = `
Object {
  "authorIds": Array [],
  "codeSizeLevel": "tiny",
  "description": "",
  "gdevelopVersion": "",
  "id": "0c9a01cf4f401ce3df54cdbc9168ba652b3b43265b3dcbbed754599247165e5a",
  "license": "MIT",
  "linkedExampleShortHeaders": Array [
    Object {
      "relation": "pixel-art-version",
      "slug": "starting-platformer-pixel",
    },
  ],
  "name": "Platformer",
  "previewImageUrls": Array [
    "https://resources.gdevelop-app.com/examples/starting-platformer/preview.png",
  ],
  "shortDescription": "Starting point for a platformer game.",
  "slug": "starting-platformer",
  "tags": Array [
    "starting-platformer",
    "Starting point",
    "Objects with animations",
    "Sounds and music",
    "Mathematical tools",
    "Objects",
    "Variables",
    "Debugger Tools",
    "Objects with effects",
    "Flippable objects",
    "Objects with opacity",
    "Platform behavior",
    "Resizable objects",
    "Scalable objects",
    "Sprite",
    "Tiled Sprite Object",
    "Tweening",
    "Button states and effects",
    "Multitouch joystick and buttons (sprite)",
    "Smooth Camera",
  ],
}
`;

exports[`exampleShortHeaders.json post build checks defaultTags 4`] = `
Object {
  "authorIds": Array [],
  "codeSizeLevel": "tiny",
  "description": "",
  "gdevelopVersion": "",
  "id": "986ba30910cf3873205796d8331f76dee507673d89e55b04415759cd39b2ebca",
  "license": "MIT",
  "linkedExampleShortHeaders": Array [
    Object {
      "relation": "non-pixel-art-version",
      "slug": "starting-platformer",
    },
  ],
  "name": "Platformer (Pixel Art)",
  "previewImageUrls": Array [
    "https://resources.gdevelop-app.com/examples/starting-platformer-pixel/preview.png",
  ],
  "shortDescription": "Starting point for a platformer game, optimized for pixel art.",
  "slug": "starting-platformer-pixel",
  "tags": Array [
    "starting-platformer-pixel",
    "Objects with animations",
    "Sounds and music",
    "Layers and cameras",
    "Mathematical tools",
    "Objects",
    "Scene",
    "Variables",
    "Debugger Tools",
    "Objects with effects",
    "Flippable objects",
    "Objects with opacity",
    "Platform behavior",
    "Resizable objects",
    "Scalable objects",
    "Sprite",
    "Tiled Sprite Object",
    "Tweening",
    "Button states and effects",
    "Multitouch joystick and buttons (sprite)",
    "Smooth Camera",
  ],
}
`;
